src/DelayId.cc

```diff
@@ -147,12 +130,7 @@ DelayId::bytesWanted(int minimum, int maximum) const
 void
 DelayId::bytesIn(int qty)
 {
-    if (! (*this))
-        return;
-
-    assert ((unsigned short)(pool() - 1) != 0xFFFF);
-
-    if (compositeId != nullptr)
+    if (*this)
         compositeId->bytesIn(qty);
```

Comment:
I expect this change to be blocked during official review because an earlier change fixing the problematic `if` statement "format" was blocked in prerequisite PR. I see three reasonable options:

* Add a new constant DelayId::mayDelay() method. Use it in Delay.cc instead of the boolean operator. The old operator will return mayDelay() result (and will still be used outside of Delay.cc). This slightly enlarges this PR scope but still moves us forward.
* Adjust this and DelayId::bytesWanted() implementation in Delay.cc to _preserve_ the existing problematic `if` statement lines. Doing so will restore some code duplication inside DelayId::bytesWanted(), unfortunately.
* Post this PR without either of the above adjustments and hope for the best.

Your call.

Reply:
I restored the original condition with 'return' at the beginning of the method - there should be no disputes about condition changing then.